### ISSUE TITLE
Broker does not support initiating registration to a newly created nameserver

### ISSUE THREAD

**Issue by stheart** [apache/rocketmq-operator#52]
Question

we use rocketmq-operator to deploy in k8s,found that when up-scale name server cluster,broker will restart one by one.

I check the broker code, it seems that broker can't refresh name server address list automatic when add new name server pod, so operator has to restart all broker.

FEATURE REQUEST

broker support register to new name server pod,don't need to restart. 
for example
1.broker provide api to refresh name server address
2.rocketmq-operator  push the new name server address to broker api when add new nameserver pod.

**Comment by liuruiyiyang** [comment]
Yes, the broker does not support hot-change the name server IP list yet (at lease in version 4.5.0), so currently our solution is to restart them one by one. And users can choose not to restart in Broker CR by setting ```allowRestart```. In fact we have some discussion on this, checkout RocketMQ PR#1458 https://github.com/apache/rocketmq/pull/1458 . We will continue to look at this issue.

**Comment by RockteMQ-AI** [comment]
**Issue Evaluation**

Category: `enhancement` | Status: **Evaluated**

**Feasibility:** feasible (requires coordinated changes in Broker + Operator)
**Scope:** Broker NameServer address refresh, Operator scaling logic
**Compatibility:** backward-compatible if Broker gracefully handles new NameServer registration

The analysis is correct: Broker currently loads NameServer addresses at startup and does not dynamically refresh them. When the Operator scales up the NameServer cluster, it must restart Brokers to pick up the new addresses.

This is a valid enhancement request. A proper fix would involve:
1. Broker supporting dynamic NameServer list refresh (e.g., via AdminTools API or file watch)
2. Operator leveraging that mechanism instead of rolling restarts

Marking as a long-term enhancement. Implementation requires cross-module coordination.

---
*Automated evaluation by RockteMQ-AI*
